nixos/modules/services/misc/fastapi-dls.nix

```diff
@@ -0,0 +1,133 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.fastapi-dls;
+  stateDir = "/var/lib/fastapi-dls";
+  dls_privkey = "${stateDir}/instance.private.pem";
+  dls_pubkey = "${stateDir}/instance.public.pem";
+  https_privkey = "${stateDir}/webserver.key";
+  https_cert = "${stateDir}/webserver.crt";
+in
+{
+  ##### interface
+  options = {
+    services.fastapi-dls = {
+      enable = lib.mkEnableOption "fastapi-dls";
+
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.fastapi-dls;
+        defaultText = lib.literalExpression "pkgs.fastapi-dls";
+        description = "Which package to use for the fastapi-dls daemon.";
+      };
+
+      listenAddress = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        description = "The IP address on which `fastapi-dls` listens.";
+      };
+
+      listenPort = lib.mkOption {
+        type = lib.types.port;
+        default = 443;
+        description = "The port on which `fastapi-dls` listens.";
+      };
+
+      dlsAddress = lib.mkOption {
+        type = lib.types.str;
+        default = cfg.listenAddress;
+        defaultText = lib.literalExpression "services.fastapi-dls.listenAddress";
+        description = ''
+          The HTTPS domain name that DLS clients should connect to.
+          Useful when you put `fastapi-dls` behind a reverse proxy.
+        '';
+      };
+
+      dlsPort = lib.mkOption {
+        type = lib.types.port;
+        default = cfg.listenPort;
+        defaultText = lib.literalExpression "services.fastapi-dls.listenPort";
+        description = "The port that DLS clients should connect to.";
+      };
+
+      openFirewall = lib.mkEnableOption "opening the firewall for `fastapi-dls`";
+    };
+  };
+
+  ##### implementation
+  config =
+    let
+      envFile = pkgs.writeText "fastapi-dls.env" ''
```

Comment:
This is not extensible. Also we should probably strip the commented things.

Reply:
The content (including comments) here is based on the [upstream env file](https://git.collinwebdesigns.de/oscar.krause/fastapi-dls/-/blob/main/.DEBIAN/env.default) and is [how the upstream configure](https://git.collinwebdesigns.de/oscar.krause/fastapi-dls/-/blob/main/.DEBIAN/fastapi-dls.service?ref_type=heads#L13) the service.

These are just environment variables though, so we can definitely use different implementation. Do you have an example in mind?

Reply:
`systemd.services.fastapi-dls.environment`?

Reply:
Switched to `systemd.services.fastapi-dls.environment` but I'm not sure how this is more extensible.